futurex_openedx_extensions/dashboard/serializers.py

```diff
@@ -166,20 +177,36 @@ def _get_arabic_name(self, obj: Any) -> str:
             arabic_full_name = ' '.join(filter(None, (arabic_first_name, arabic_last_name)))
         return (arabic_full_name or '').strip()
 
-    @staticmethod
-    def _get_profile_field(obj: get_user_model, field_name: str) -> Any:
+    def _get_profile_field(self: Any, obj: get_user_model, field_name: str) -> Any:
         """Get the profile field value."""
-        return getattr(obj.profile, field_name) if hasattr(obj, 'profile') and obj.profile else None
+        user = self._get_user(obj)
+        return getattr(user.profile, field_name) if hasattr(user, 'profile') and user.profile else None
 
-    @staticmethod
-    def _get_extra_field(obj: get_user_model, field_name: str) -> Any:
+    def _get_extra_field(self: Any, obj: get_user_model, field_name: str) -> Any:
         """Get the extra field value."""
-        return getattr(obj.extrainfo, field_name) if hasattr(obj, 'extrainfo') and obj.extrainfo else None
+        user = self._get_user(obj)
+        return getattr(user.extrainfo, field_name) if hasattr(user, 'extrainfo') and user.extrainfo else None
 
     def get_user_id(self, obj: get_user_model) -> Any:  # pylint: disable=no-self-use
         """Return user ID."""
         return obj.id
 
+    def get_email(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return self._get_user(obj).email  # type: ignore
+
+    def get_username(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return self._get_user(obj).username  # type: ignore
+
+    def get_date_joined(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return str(self._get_user(obj).date_joined)  # type: ignore
+
+    def get_last_login(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return str(self._get_user(obj).last_login)  # type: ignore
```

Comment:
don't trust `str` when converting date and time. At least it is not timezone-aware
```suggestion
    def get_date_joined(self, obj: Any) -> str:
        date_joined = self._get_user(obj).date_joined
        return date_joined.isoformat() if date_joined else None

    def get_last_login(self, obj: Any) -> str:
        last_login = self._get_user(obj).last_login
        return last_login.isoformat() if last_login else None
```